process/block/cutoff/blockProcessingCutoffHandler_test.go

```diff
@@ -0,0 +1,254 @@
+		require.Equal(t, "invalid block processing cutoff mode, provided value=invalid", err.Error())
+		require.Nil(t, b)
+	})
+
+	t.Run("invalid cutoff trigger - should error", func(t *testing.T) {
+		t.Parallel()
+
+		cfg := config.BlockProcessingCutoffConfig{
+			Enabled:       true,
+			Mode:          "pause",
+			CutoffTrigger: "invalid",
+		}
+		b, err := NewBlockProcessingCutoffHandler(cfg)
+		require.Equal(t, "invalid block processing cutoff trigger, provided value=invalid", err.Error())
+		require.Nil(t, b)
+	})
+
+	t.Run("should work", func(t *testing.T) {
+		t.Parallel()
+
+		cfg := config.BlockProcessingCutoffConfig{
+			Enabled:       true,
+			Mode:          "pause",
+			CutoffTrigger: "epoch",
+		}
+		b, err := NewBlockProcessingCutoffHandler(cfg)
+		require.NoError(t, err)
+		require.False(t, check.IfNil(b))
+	})
+}
+
+func TestBlockProcessingCutoffHandler_HandlePauseBackoff(t *testing.T) {
+	t.Parallel()
+
+	t.Run("bad config - should skip", func(t *testing.T) {
+		t.Parallel()
+
+		defer func() {
+			r := recover()
+			require.Nil(t, r)
+		}()
+
+		cfg := config.BlockProcessingCutoffConfig{
+			Enabled:       false,
+			Mode:          "pause",
+			CutoffTrigger: "nonce",
+		}
+		b, err := NewBlockProcessingCutoffHandler(cfg)
+		require.NoError(t, err)
+
+		b.HandlePauseCutoff(nil)
+		b.HandlePauseCutoff(&block.MetaBlock{})
+		b.config.CutoffTrigger = common.BlockProcessingCutoffModeProcessError
+		b.HandlePauseCutoff(&block.MetaBlock{})
+	})
+
+	t.Run("pause via round - should work", func(t *testing.T) {
+		t.Parallel()
+
+		cfg := config.BlockProcessingCutoffConfig{
+			Enabled:       true,
+			Mode:          common.BlockProcessingCutoffModePause,
+			CutoffTrigger: string(common.BlockProcessingCutoffByRound),
+			Value:         20,
+		}
+		b, err := NewBlockProcessingCutoffHandler(cfg)
+		require.NoError(t, err)
+
+		b.HandlePauseCutoff(&block.MetaBlock{Round: 19}) // not the desired round
+
+		done := make(chan struct{})
+		go func() {
+			b.HandlePauseCutoff(&block.MetaBlock{Round: 20})
+			done <- struct{}{}
+		}()
+
+		select {
+		case <-done:
+			require.Fail(t, "should have not advanced")
+		case <-time.After(100 * time.Millisecond):
+		}
+	})
+
+	t.Run("pause via nonce - should work", func(t *testing.T) {
+		t.Parallel()
+
+		cfg := config.BlockProcessingCutoffConfig{
+			Enabled:       true,
+			Mode:          common.BlockProcessingCutoffModePause,
+			CutoffTrigger: string(common.BlockProcessingCutoffByNonce),
+			Value:         20,
+		}
+		b, err := NewBlockProcessingCutoffHandler(cfg)
+		require.NoError(t, err)
+
+		b.HandlePauseCutoff(&block.MetaBlock{Nonce: 19}) // not the desired round
+
+		done := make(chan struct{})
+		go func() {
+			b.HandlePauseCutoff(&block.MetaBlock{Nonce: 20})
+			done <- struct{}{}
+		}()
+
+		select {
+		case <-done:
+			require.Fail(t, "should have not advanced")
+		case <-time.After(100 * time.Millisecond):
+		}
+	})
+
+	t.Run("pause via epoch - should work", func(t *testing.T) {
```

Comment:
optional: the duplicated code from these should work tests can be reduced(similar on `process error via ...`):
```
...
	t.Run("pause via round - should work", testHandlePauseCutoff(string(common.BlockProcessingCutoffByRound)))
	t.Run("pause via nonce - should work", testHandlePauseCutoff(string(common.BlockProcessingCutoffByNonce)))
	t.Run("pause via epoch - should work", testHandlePauseCutoff(string(common.BlockProcessingCutoffByEpoch)))
}

func testHandlePauseCutoff(trigger string) func(t *testing.T) {
	return func(t *testing.T) {
		t.Parallel()

		cfg := config.BlockProcessingCutoffConfig{
			Enabled:       true,
			Mode:          common.BlockProcessingCutoffModePause,
			CutoffTrigger: trigger,
			Value:         20,
		}
		b, err := NewBlockProcessingCutoffHandler(cfg)
		require.NoError(t, err)

		b.HandlePauseCutoff(&block.MetaBlock{
			Epoch: 19, // not the desired epoch
			Nonce: 19, // not the desired nonce
			Round: 19, // not the desired round
		})

		done := make(chan struct{})
		go func() {
			b.HandlePauseCutoff(&block.MetaBlock{
				Epoch: 20,
				Nonce: 20,
				Round: 20,
			})
			done <- struct{}{}
		}()

		select {
		case <-done:
			require.Fail(t, "should have not advanced")
		case <-time.After(time.Millisecond):
		}
	}
}
```